lib/iris/fileformats/netcdf.py

```diff
@@ -1299,6 +1299,14 @@ def _cf_coord_identity(self, coord):
             elif coord.standard_name == "longitude":
                 units = 'degrees_east'
 
+        elif isinstance(coord.coord_system, iris.coord_systems.RotatedGeogCS):
+            units = 'degrees'
+
+        elif isinstance(coord.coord_system,
+                        (iris.coord_systems.TransverseMercator,
+                         iris.coord_systems.LambertAzimuthalEqualArea)):
+            units = 'm'
+
```

Comment:
This block was removed in PR #1943 but I don't agree that it should have been removed. 
[Section 4.1 and 4.2](http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html#latitude-coordinate) of the cf conventions says:

> Coordinates of latitude with respect to a rotated pole should be given units of degrees

also 'm' is a perfectly fine unit (see the example in [this section](http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html#_two_dimensional_latitude_longitude_coordinate_variables)